DROOLS-2522: [DMN Designer] Palette aesthetics

manstis opened the pull request:
See https://issues.jboss.org/browse/DROOLS-2522 specifically https://issues.jboss.org/browse/DROOLS-2522?focusedCommentId=13618379&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13618379. This PR makes the Palette look nicer. No change to icons; just its layout.

@danielzhe Would you be able to have a quick check on IE11? I can talk you through what you'd need to compile/run. Is that OK?

@wmedvede Added you since you know a bit about these palettes ;-)

danielzhe commented:
@danielzhe Sure. I'm on it.

danielzhe commented:
@manstis Sure! I'm on it.

wmedvede commented on kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/palette/categories/items/DefinitionPaletteItemWidgetViewImpl.css:
```diff
@@ -1,18 +1,12 @@
-
 .kie-palette-item-anchor-spacer {
-    display:inline-block !important; /* IE11 */
+    display: inline-block;
```
@danielzhe I would same the same as @manstis . Please check that both BPMN and DMN editor palettes works well in all browsers. Specially IE11, and Edge too, since it took considerable time to fine tune this stuff for making it work well in all browsers.

manstis replied:
@wmedvede The use of `!important` broke the spacing of a Palette generated from `CollapsedPaletteDefinitionBuilder` (it lead to extra space between the palette items vertically). I tested my changes on BPMN and DMN on Chrome and all was good... but as we know IE11 can be somewhat different :-(

wmedvede replied:
Hi @manstis I've pointed to this particular line of code just for highlighting the fact that there are "tunnings" added just because of the compatibility. I can't say I'm proud of some pieces, but was the way to make it work... Let's say that this area is a sort of "browser sensitive" stuff. So I strongly suggest that we do the browser compatibilitiy tests good enough at least for this area. And yes, some of the "shit" added there comes from me :).

jomarko commented:
I am not able to check currently on edge as have no win10 just windows server, so checked IE11, Firefox and Chrome. The only difference is on IE11 for stunner, see:
- IE11
  - ![screenshot from 2018-10-01 10-07-42](https://user-images.githubusercontent.com/8044780/46277001-1d123700-c562-11e8-874a-12f2da0ea1c8.png)
- Chrome/Firefox
  - ![screenshot from 2018-10-01 10-07-54](https://user-images.githubusercontent.com/8044780/46277015-28656280-c562-11e8-8a46-8a94554ca780.png)

manstis commented:
Thanks @jomarko 

So rather than tinker with the risky CSS for the `DefinitionPaletteItemWidgetViewImpl` I'll make the "Collapsed Palette" use a completely new "Palette Item Widget".. This PR is going to change.

jomarko commented:
Attaching Edge, In low resolution but seems fine:
![screenshot from 2018-10-01 11-51-42](https://user-images.githubusercontent.com/8044780/46281941-7b461680-c570-11e8-8e4d-104998f8b39e.png)

manstis commented:
@wmedvede @jomarko @danielzhe Hi, I've changed this PR to leave the existing `DefinitionPaletteItemWidget` unchanged (as it's used by BPMN for it's categorised palette). Instead I've now added a new Palette Widget element used solely by the `CollapsedPaletteDefinitionBuilder` with its own classes and HTML etc. Could you please re-check both BPMN and DMN on various browsers, including IE11.

danielzhe commented:
@manstis Can you provide me more info about how can I run this to test? Is in DMN design? Thanks.

manstis commented:
Hi @danielzhe you can test this by following the "Command line instructions" above to fetch this PR locally (on top of `kie-wb-common`); then perform a `mvn clean install -DskipTests -Dgwt.compiler.skip=true` on `kie-wb-common` root. Once that completes you can `cd ./kie-wb-common/kie-wb-common-dmn/kie-wb-common-dmn-webapp` and perform `mvn clean process-resources gwt:run`. When the GWT "launch" dialog shows "Launch default browser" button (next to the application URL) click it, type `admin/admin` and finally "Create DMN" from the menubar. The _palette_ should be a column of vertical buttons that you can drag from the palette to the canvas. If any of this is unclear please just ping me on IRC and we'll go through it together. Thanks, Mike

jomarko commented on kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/palette/collapsed/CollapsedDefinitionPaletteItemWidgetViewImpl.java:
```diff
@@ -0,0 +1,93 @@
+
+    private static boolean isEmpty(final String s) {
+        return null == s || s.trim().length() == 0;
+    }
```
I think we shouldn't introduce the same method again. Output of grep below.
```
kie-wb-common-stunner-extensions/kie-wb-common-stunner-svg/kie-wb-common-stunner-svg-gen/src/main/java/org/kie/workbench/common/stunner/svg/gen/translator/impl/SVGDocumentTranslatorImpl.java:        return null == s || s.trim().length() == 0;
kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/palette/collapsed/CollapsedDefinitionPaletteItemWidgetViewImpl.java:        return null == s || s.trim().length() == 0;
kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/palette/categories/items/DefinitionPaletteItemWidgetViewImpl.java:        return null == s || s.trim().length() == 0;
kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/workitem/service/WorkItemDefinitionRemoteService.java:        return null == s || s.trim().length() == 0;
kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/workitem/WorkItemDefinitionParser.java:        return null == s || s.trim().length() == 0;

```

jomarko commented:
@manstis manual check passed, please see my comment to code and I can approve.

manstis commented:
@jomarko Duplicate `isEmpty(..)` method moved to utility class in new commit.

wmedvede commented:
@manstis 
I've double checked locally that DMN and BPMN palettes are working well in Firefox, Chrome, Edge and IE11, for both kie-wb-common-dmn-webapp.war and kie-wb-7.13.0-SNAPSHOT-wildfly11.war apps.
Thank you!

danielzhe commented:
Sorry for the delay, @manstis . My SSD just got trashed today and I'm without Windows here. Thanks @wmedvede for checked it!
